debugtools/DDR_VM/src/com/ibm/j9ddr/J9DDRClassLoader.java

```diff
@@ -63,32 +62,38 @@ private static boolean shouldGeneratePointerClasses(StructureReader reader) {
 			return false;
 		}
 
-		// Beginning with VM version 29, pointers classes are generated
-		// if "generate.pointers" is true in dynamic.properties.
-
-		String packageName = reader.getPackageName(PackageNameType.POINTER_PACKAGE_SLASH_NAME);
-		String resourceName = '/' + packageName + "dynamic.properties";
-		InputStream stream = J9DDRClassLoader.class.getResourceAsStream(resourceName);
-
-		if (stream == null) {
-			return false;
-		}
-
-		Properties dynamic = new Properties();
-
 		try {
-			dynamic.load(stream);
-		} catch (IOException e) {
+			// Get pointer sizes in bytes (rounding up for 31-bit s390 VMs).
+			int hostPointerSize = (Integer.getInteger("sun.arch.data.model", 0) + 7) / 8;
+			int corePointerSize = reader.getSizeOfUDATA();
+
+			if (hostPointerSize == corePointerSize) {
+				// The VM that produced the core file has the same size pointers
+				// as this VM: we can use the pointers classes in j9ddr.jar.
+				return false;
```

Comment:
I'd like to find a way to verify this check is working as designed. Perhaps some debug output that can be enabled to indicate if pointer classes are being generated at runtime or not.

Reply:
Running `jdmpview` with the option
```
-J-Xtrace:none,maximal=mt,methods=com/ibm/j9ddr/J9DDRClassLoader.shouldGeneratePointerClasses(),output=ddr.log
```
will capture the return value of this method. I don't expect we'll need to check so often as to make a more convenient solution worth the effort.